volatility3/framework/plugins/windows/strings.py

```diff
@@ -149,9 +178,9 @@ def generate_mapping(
             for mapval in layer.mapping(0x0, layer.maximum_address, ignore_errors=True):
                 offset, _, mapped_offset, mapped_size, maplayer = mapval
                 for val in range(mapped_offset, mapped_offset + mapped_size, 0x1000):
-                    cur_set = reverse_map.get(val >> 12, set())
-                    cur_set.add(("kernel", offset))
-                    reverse_map[val >> 12] = cur_set
+                    cur_set = reverse_map.get(val, set())
+                    cur_set.add(("kernel", val))
+                    reverse_map[offset] = cur_set
```

Comment:
This appears to have changed the index from what had been `val` to `offset`.  I assume that intentional but it looks like it might potentially leave two entries in the reverse_map.  Since this code appears to just be adding to the members of the set, taking the set from one index and putting it back in a (potentially) different index, seems like a bad idea...

Reply:
Yes they were listing the wrong way round, you can see this in the explanation i left in the issue thread comparing the 2.6 and 3 plugins these two values were being put in the wrong way round, I just switched them